lib/deep-required/index.ts

```diff
@@ -18,6 +20,12 @@ export type DeepRequired<Type> = Type extends Error
   ? WeakSet<DeepRequired<Values>>
   : Type extends Promise<infer Value>
   ? Promise<DeepRequired<Value>>
+  : Type extends AnyArray<infer Values>
```

Comment:
It would be really nice to avoid Tuples leaking through to the left-hand side of the conditional so that we can let them drop all the way down to the type (now) on line 30 and reduce the repetition a bit.

Reply:
I think it's my bad that `AnyArray` is used in these utility types, you can simply use `ReadonlyArray<infer Values>` or `readonly (infer Values)[]`

For tuples, there's a utility type [IsTuple](https://github.com/ts-essentials/ts-essentials/tree/master/lib/is-tuple), you can use it to filter it and process only arrays at this point

Reply:
Done :+1: